### PR TITLE
[local_auth] Convert native unit tests to Swift

### PR DESCRIPTION
**Pull request by stuartmorgan**
Converts native unit tests from Objective-C to Swift, as a first step toward an eventual plugin conversion. Since OCMock usage was removed in a previous PR, the tests are converted essentially directly (the rewrites were largely mechanical syntax replacement), without any changes needed to production code.

There are a few places where interacting with the Pigeon-generated Obj-C is somewhat awkward because the NSError signature isn't auto-converting to `throw` in cases where we use `NSNumber` wrapping, but all that will get cleaned up when we switch the plugin over to Swift Pigeon generation and have idiomatic Swift APIs that the tests will be calling instead.

Fixes a couple of false positives in the repo tooling surfaced by this PR:
- `darwin/Tests/` wasn't recognized as a test directory.
- Swift tests weren't recognized as exempt from requiring Swift entries in the podspec.

Part of https://github.com/flutter/flutter/issues/119104

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I [linked to at least one issue that this PR fixes] in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [ ] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[linked to at least one issue that this PR fixes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Comment by stuartmorgan** [comment]
That `check_podspec` failure is a false positive; I'll just need to update the repo tooling to fix that.

File: packages/local_auth/local_auth_darwin/darwin/Tests/FLALocalAuthPluginTests.swift [hunk 1]
```diff
@@ -0,0 +1,403 @@
...
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import Flutter
+import XCTest
+
+@testable import local_auth_darwin
+
+// Set a long timeout to avoid flake due to slow CI.
+let timeout: TimeInterval = 30.0
```
Comment (hellohuanlin) [inline packages/local_auth/local_auth_darwin/darwin/Tests/FLALocalAuthPluginTests.swift:11]: nit: can be `fileprivate` (unless you want to make this `timeout` available in the whole test module)

File: packages/local_auth/local_auth_darwin/darwin/Tests/FLALocalAuthPluginTests.swift [hunk 2]
```diff
@@ -0,0 +1,403 @@
...
+  var contexts: [FLADAuthContext]
+  init(contexts: [FLADAuthContext]) {
+    self.contexts = contexts
+  }
+
+  func createAuthContext() -> FLADAuthContext {
+    XCTAssert(self.contexts.count > 0, "Insufficient test contexts provided")
+    let context = self.contexts.first!
+    contexts.remove(at: 0)
+    return context
```
Comment (hellohuanlin) [inline packages/local_auth/local_auth_darwin/darwin/Tests/FLALocalAuthPluginTests.swift:24]: nit: 

```
return self.contexts.removeFirst()
```

File: packages/local_auth/local_auth_darwin/darwin/Tests/FLALocalAuthPluginTests.swift [hunk 3]
```diff
@@ -0,0 +1,403 @@
...
+  var expectBiometrics = false
+  /// The error to return from canEvaluatePolicy.
+  var canEvaluateError: NSError?
+  /// The value to return from evaluatePolicy:error:.
+  var evaluateResponse = false
+  /// The error to return from evaluatePolicy:error:.
+  var evaluateError: NSError?
+
+  // Overridden as read-write to allow stubbing.
+  var biometryType: LABiometryType = LABiometryType.none
```
Comment (hellohuanlin) [inline packages/local_auth/local_auth_darwin/darwin/Tests/FLALocalAuthPluginTests.swift:43]: nit: type can be inferred 

```
  var biometryType: LABiometryType = .none
```

File: packages/local_auth/local_auth_darwin/darwin/Tests/FLALocalAuthPluginTests.swift [hunk 4]
```diff
@@ -0,0 +1,403 @@
...
+    }
+  }
+}
+
+// MARK: -
+
+class FLALocalAuthPluginTests: XCTestCase {
+
+  override func setUp() {
+    continueAfterFailure = false
```
Comment (hellohuanlin) [inline packages/local_auth/local_auth_darwin/darwin/Tests/FLALocalAuthPluginTests.swift:79]: curious why? (also maybe add comment)

File: packages/local_auth/local_auth_darwin/darwin/Tests/FLALocalAuthPluginTests.swift [hunk 5]
```diff
@@ -0,0 +1,403 @@
...
+import Flutter
+import XCTest
+
+@testable import local_auth_darwin
+
+// Set a long timeout to avoid flake due to slow CI.
+let timeout: TimeInterval = 30.0
+
+/// A context factory that returns preset contexts.
+class StubAuthContextFactory: NSObject, FLADAuthContextFactory {
```
Comment (hellohuanlin) [inline packages/local_auth/local_auth_darwin/darwin/Tests/FLALocalAuthPluginTests.swift:14]: nit: can be `final class` (unless we intend to subclass it).

**Review by hellohuanlin — APPROVED** [review]

File: packages/local_auth/local_auth_darwin/darwin/Tests/FLALocalAuthPluginTests.swift (hunk 1 above)
Comment (stuartmorgan) [reply packages/local_auth/local_auth_darwin/darwin/Tests/FLALocalAuthPluginTests.swift]: Done.

File: packages/local_auth/local_auth_darwin/darwin/Tests/FLALocalAuthPluginTests.swift (hunk 5 above)
Comment (stuartmorgan) [reply packages/local_auth/local_auth_darwin/darwin/Tests/FLALocalAuthPluginTests.swift]: Done.

File: packages/local_auth/local_auth_darwin/darwin/Tests/FLALocalAuthPluginTests.swift (hunk 2 above)
Comment (stuartmorgan) [reply packages/local_auth/local_auth_darwin/darwin/Tests/FLALocalAuthPluginTests.swift]: Nice, I didn't realize Swift arrays had fixed some of the annoying API limitations of `NSArray` (like not having a remove-and-return method).

File: packages/local_auth/local_auth_darwin/darwin/Tests/FLALocalAuthPluginTests.swift (hunk 3 above)
Comment (stuartmorgan) [reply packages/local_auth/local_auth_darwin/darwin/Tests/FLALocalAuthPluginTests.swift]: Fixed throughout.

File: packages/local_auth/local_auth_darwin/darwin/Tests/FLALocalAuthPluginTests.swift (hunk 4 above)
Comment (stuartmorgan) [reply packages/local_auth/local_auth_darwin/darwin/Tests/FLALocalAuthPluginTests.swift]: No clue, it looks like it dates back to my adding the original test file. Copypasta from an XCUITest maybe? Removed.

**Comment by stuartmorgan** [comment]
Changelog override: only affects tests.

**Pull request merged**